### DISCUSSION TITLE
Problem with delete row automatiuon with Budibase DB

### DISCUSSION THREAD

**Discussion by BismaIftekhar** [Budibase/budibase#8152] [Help] [answered]
Hello everyone,

I have two tables with same and different columns in both tables. 
1) switches_test
2) database_test

I want to create an automation if a row gets deleted in one table, then that row also gets deleted from the other table.

I have tried with the following but none seem to work. 

{{ trigger.row }}
{{ trigger.row._id}}
{{ trigger.row.id}}
{{ trigger.row[Auto ID] }}

![automation_error](https://user-images.githubusercontent.com/96243721/194262961-fa948a12-dbb9-4bfa-9225-87d3cbff7ac6.png)

Can anyone help please?
Thank you.

**Comment by andz-bb** [comment, marked as answer]
Hey @BismaIftekhar 

for this one you'll need to get the field ID `_id` from database_test that matches the row you're deleting in switch_test. 

If your rows are in Budibase DB and have a one to many relationship set up you could do something like `{{ trigger.row.database.[0] }}` to get the linked row _id from the other table.

If they're not linked with a relationship you would need to add a Query Rows step before the Delete Row action. You will be able to query the database_test table to get the row that matches the row you deleted. You can then use the Step 1 output to set the Row ID in the Delete Row action. I passed this binding in and it worked ok for me: `{{ steps.1.rows.[0]._id }}`

If you have any further issues or questions on this let me know and I'll be happy to help 😊

**Reply by BismaIftekhar** [reply]
Thank you so much @andz-bb for taking out time for this.

Just for information. The tables do not have a relationship and do not have matching ids. The name,location, IP address is same in both tables but the ids are different. So I want that if a row with name switch-A , IP address 0.0.0.0 (for example) gets deleted from the switch_test table then that row also gets deleted from database_test table.

You mentioned "...you will be able to query the database_test table to get the row that matches the row you deleted" how do I set this up?I am assuming I´ll have to set this is filter. 

Sorry for bothering you again but would be grateful if you could help. Thank you, again.

**Reply by andz-bb** [reply]
hey @BismaIftekhar 

sorry for being light on detail on that part of my response. I've attached a full app export with a demo of what I'm talking about.
[delete automation-export-1665067965641.txt](https://github.com/Budibase/budibase/files/9725843/delete.automation-export-1665067965641.txt)

Here's a screenshot of my automation screen. 

<img width="414" alt="Screenshot 2022-10-06 at 15 55 30" src="https://user-images.githubusercontent.com/110921612/194346564-99c9ed85-6a00-4c00-9b48-12d74a923b0d.png">

In my Query Rows action I have set up my filters like so to return only the rows where the IP column in database_test matches the IP from the trigger row. Assuming it's a one-to-one relationship you need to limit the query to one result. Query Rows returns an array of rows so in the Delete Row action when providing the ID you want to take the first object in the array and then get the ID from it.

<img width="1054" alt="Screenshot 2022-10-06 at 15 56 06" src="https://user-images.githubusercontent.com/110921612/194346723-099d3818-e20b-4045-a2d4-bc4816015a0d.png">

Hope this clears things up for you, let me know if you have any other questions 👍

**Reply by BismaIftekhar** [reply]
Hey @andz-bb,
My apologies for the late response. Thank you so much for such an elaborate response. It did help! (Just for info: I did not limit the query result to 1.There is no relationship between the tables)
Thanks again:)
